### PR TITLE
report: new score display, metrics display, rich tooltips

### PR DESCRIPTION
**Pull request by paulirish**
### Score display

New gauges, done in SVG instead of crazytime css. Added in the score scale. 

<details>
	<summary>Oh, they animate, too. :) </summary>

(easy to remove if everyone h8s it.)

</details>

#### Before:
image
#### After:
image

And check out the difference in CC layers:

### Metrics display

Not 100% finished but getting there.

image

I've added in a rich tooltip system to support this and other things.. here's how it works:

```html
<div class="tooltip-boundary">  <!-- any hover within this element will show the tooltip -->
	<!-- stuff that you see.. -->
    <div class="tooltip"> ... </div> <!-- hidden by default. can contain w/e -->
</div>
```

#ref #5008

### CODE REVIEW

**Comment by vinamratasingal** [comment]
This looks AMAZING! CC: @hwikyounglee 

I especially love the initial animation, makes the UI look #snazzy. Thanks @paulirish !

File: lighthouse-core/report/v2/report-styles.css [hunk 1]
```diff
@@ -362,78 +362,82 @@ span.lh-node:hover {
...
+  background-size: contain;
+  display: inline-block;
+  vertical-align: text-bottom;
+  margin-left: calc(var(--body-font-size) / 2);
 }
 
-.lh-timeline-metric--fail .lh-timeline-metric__value {
-  color: var(--fail-color);
+.lh-perf-metric--pass .lh-perf-metric__value span::after {
+  background: url('data:image/svg+xml;utf8,<svg version="1.1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 130.2 130.2"><circle class="path circle" fill="#73AF55" cx="65.1" cy="65.1" r="62.1"/><polyline class="path check" fill="none" stroke="white" stroke-width="16" stroke-linecap="round" stroke-miterlimit="10" points="100.2,40.2 51.5,88.8 29.8,67.5 "/></svg>') no-repeat 50% 50%;
```
[the remark this answers is not in the document]
Comment (paulirish) [reply lighthouse-core/report/v2/report-styles.css]: great idea. will do when i update to hwi's new assets.

File: lighthouse-core/report/v2/renderer/performance-category-renderer.js [hunk 2]
```diff
@@ -119,38 +115,37 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
...
-      if (typeof audit.result.rawValue === 'number' && audit.result.rawValue) {
-        perfTimelineScale = Math.max(perfTimelineScale, audit.result.rawValue);
-      }
+
+    // Metrics
+    const keyMetrics = metricAudits.filter(a => a.weight >= 3);
+    const otherMetrics = metricAudits.filter(a => a.weight < 3);
+
+    const metricsBoxesEl = this.dom.createChildOf(metricAuditsEl, 'div', 'lh-metrics-container');
+    const keyBox = this.dom.createChildOf(metricsBoxesEl, 'div', 'lh-metrics lh-metrics__key');
```
[the remark this answers is not in the document]
Comment (brendankenny) [reply lighthouse-core/report/v2/renderer/performance-category-renderer.js]: maybe just adding in `Metrics` is enough? like `keyMetricsBox`?

File: lighthouse-core/report/v2/templates.html [hunk 3]
```diff
@@ -35,22 +45,14 @@
...
-      <div class="lh-timeline-metric__value"><!-- fill me --></div>
-      <details class="lh-timeline-metric__details lh-expandable-details">
-        <summary class="lh-timeline-metric__summary lh-expandable-details__summary">
-          <span class="lh-timeline-metric__title"><!-- fill me --></span>
-          <div class="lh-toggle-arrow" title="See audits"></div>
-        </summary>
-        <div class="lh-timeline-metric__description"><!-- fill me --></div>
-      </details>
-    </div>
+<template id="tmpl-lh-perf-metric">
```
Comment (brendankenny) [inline lighthouse-core/report/v2/templates.html]: update comment above?

File: lighthouse-core/test/results/sample_v2.json [hunk 4]
```diff
@@ -5347,6 +5347,6 @@
     }
   },
   "timing": {
-    "total": 830
+    "total": 1142
```
Comment (brendankenny) [inline lighthouse-core/test/results/sample_v2.json]: no need to update?

**Pull request merged**